catalog/lib/theme/app_theme.dart

```diff
@@ -7,19 +8,41 @@ import 'package:catalog/theme/app_dimensions.dart';
 late AppColors _colors;
 late AppDimens _dimensions;
 late AppTextStyles _styles;
+late AppButtons _buttons;
 
 class AppTheme {
   static ThemeData provideAppTheme(BuildContext buildContext) {
     // It can be changed based on the device
     _styles = AppTextStyles.getAppStyles();
     _dimensions = AppDimens.getDimensions();
     _colors = AppColors.getColorScheme();
+    _buttons = AppButtons.getButtonTheme();
 
     return ThemeData(
       primaryColor: _colors.primary,
       colorScheme: _colors,
-      textTheme: _styles.getThemeData(),
-      primaryTextTheme: _styles.getThemeData(),
+      filledButtonTheme: FilledButtonThemeData(
+        style: _buttons.filledButton,
+      ),
+      outlinedButtonTheme: OutlinedButtonThemeData(
+        style: _buttons.outlineButton,
+      ),
+      textButtonTheme: TextButtonThemeData(
+        style: _buttons.textButton,
+      ),
+      scaffoldBackgroundColor: _colors.primary.shade200,
+      textTheme: _styles.getThemeData().apply(
+        bodyColor: _colors.textColor.shade100,
```

Comment:
textColor.shade100 is pretty close to white, so all text widgets will have white text color, maybe use only _colors.textColor

```suggestion
        bodyColor: _colors.textColor,
```

<img width="500" alt="image" src="https://github.com/xmartlabs/flutter-template/assets/50491531/48b5d0f7-156d-4b9e-a4ee-eb36e34e828f">